File: cpp/basix/maps.cpp
// Copyright (c) 2021 Matthew Scroggs
// FEniCS Project
// SPDX-License-Identifier:    MIT

#include "maps.h"
#include <map>
#include <stdexcept>

//-----------------------------------------------------------------------------
const std::string& basix::maps::type_to_str(maps::type type)
{
  static const std::map<maps::type, std::string> type_to_name
      = {{maps::type::identity, "identity"},
         {maps::type::covariantPiola, "covariant Piola"},
         {maps::type::contravariantPiola, "contravariant Piola"},
         {maps::type::doubleCovariantPiola, "double covariant Piola"},
         {maps::type::doubleContravariantPiola, "double contravariant Piola"}};

  auto it = type_to_name.find(type);
  if (it == type_to_name.end())
    throw std::runtime_error("Can't find type");

  return it->second;
}
//-----------------------------------------------------------------------------

File: python/basix/maps.py
# Copyright (C) 2023-2024 Matthew Scroggs and Garth N. Wells
#
# This file is part of Basix (https://www.fenicsproject.org)
#
# SPDX-License-Identifier:    MIT
"""Maps."""

from basix._basixcpp import MapType

__all__ = ["string_to_type"]


def string_to_type(mapname: str) -> MapType:
    """Convert a string to a Basix MapType.

    Args:
        mapname: Name of the map as a string.

    Returns:
        The map type.
    """
    return MapType[mapname]
